L42-metadata-flags.md

```diff
@@ -0,0 +1,140 @@
+> 
+> gRPC implementations MAY provide a per-RPC option to not fail RPCs as a result of the channel being in TRANSIENT_FAILURE state. Instead, the implementation queues the RPCs until the channel is READY. This is known as "wait for ready." The RPCs SHOULD still fail before READY if there are unrelated reasons, such as the channel is SHUTDOWN or the RPC's deadline is reached.
+> 
+> From https://github.com/grpc/grpc/blob/master/doc/wait-for-ready.md 
+
+### Use cases for “Wait For Ready”
+
+When developers spin up gRPC clients and servers in the same time, it is very like to fail first couple RPC calls due to unavailability of the server. If developers failed to prepare for this situation, the result can be catastrophic. But with “Wait For Ready” semantics, developers can initialize the client and server in any order, especially useful in testing.
+
+Also, developers may ensure the server is up before starting client. But in some cases like transient network failure may result in a temporary unavailability of the server. With “Wait For Ready” semantics, those RPC calls will automatically wait until the server is ready to accept incoming requests.
+
+### Current "Wait For Ready" behavior
+
+After the channel instance is instantiated, it will attempt to connect to the server. Even if the connection attempt is failed, the gRPC client won’t be blocked. If the first connection attempt failed, any gRPC requests immediately after will fail with StatusCode.UNAVAILABLE.
+
+But there is a Python-only function to simulate the “Wait For Ready” mechanism. It is a class named `_ChannelReadyFuture` in `_utilities.py`. It will create a `Future` object to enable the client to block till the channel is ready. It works just like the “Wait For Ready” mechanism with Pythonic handling approach, but it has three drawbacks:
+
+1. Usage is different from other languages like C++, Golang, Java. It may create inconsistency usage across language and additional cognitive cost for migrating from/to Python.
+1. Considered technical debt. More than 100 lines of code implementing the same function which C-Core already provided.
+1. It only account for the first connection attempt. Instead, C-Core “Wait For Ready” mechanism allows every RPC call to be blocked if server is unavailable.
+
+Strictly speaking, the `_ChannelReadyFuture` works in most cases. And it is already used in many Google internal projects. So this interface has to be maintained in the near future.
+
+### Desired “Wait For Ready” behavior
+After setting “Wait For Ready” option to the channel construction method, the channel will automatically enable this mechanism for all the outgoing RPC calls. NOT only the first one. This usage will be consistent with other language’s implementation.
+
+Also, by adding per RPC variable, the client can customize each RPC call to leverage the “Wait For Ready” mechanism. This can give Python a little advantage over other languages.
```

Comment:
This would negate the first point above where we want to avoid inconsistency between languages. Per-call enhancement should be left for later and done across other languages at the same time.